### ISSUE TITLE
.process() and .addJob() at the same time ???

### ISSUE THREAD

**Issue by GabrielLaTorre** [OptimalBits/bull#1890]
#### I am using Bull in a Sails project. I am going to require some services that perform external tasks and that is why I use the queues.
Testing, I managed to build a function which creates the .process () and performs the addJob () at the same time.
Will this affect the performance of my app? is there any downside to having "multiple processes"?

```
addProcessAndJob: async (function, ...args) => {
        _queue.process(async (job) => {
            return await function(...job.opts.args)
        })
        await _queue.add({ args })
    }
}
```
In the previous code "function" would be the task to be sent from another controller along with some necessary args.
Simply put, the code works, my question is whether it can affect performance.

Greetings!

**Comment by manast** [comment]
I am surprised that code works at all since bull throws an exception if you define the process function more than once. So, no, the process function should not be defined every time a job is added.
